alphabet/alphabet.go

```diff
@@ -0,0 +1,50 @@
+package alphabet
+
+import "fmt"
+
+type Alphabet struct {
+	symbols  []string
+	encoding map[interface{}]int
+}
+
+type AlphabetError struct {
```

Comment:
I may be a little naive here. Is it common pattern to define an accompanying error struct along with the struct itself?